packages/web-components/fast-foundation/src/disclosure/disclosure.template.ts

```diff
@@ -1,24 +1,28 @@
 import { ElementViewTemplate, html, ref } from "@microsoft/fast-element";
-import type { FASTDisclosure } from "./disclosure.js";
+import { endSlotTemplate, startSlotTemplate } from "../patterns/index.js";
+import type { DisclosureOptions, FASTDisclosure } from "./disclosure.js";
 
 /**
  * The template for the {@link @microsoft/fast-foundation#FASTDisclosure} component.
  * @public
  */
-export function disclosureTemplate(): ElementViewTemplate<FASTDisclosure> {
+export function disclosureTemplate(
+    options: DisclosureOptions = {}
+): ElementViewTemplate<FASTDisclosure> {
     return html<FASTDisclosure>`
-        <details class="disclosure" ${ref("details")}>
+        <details class="details" part="details" ${ref("details")}>
             <summary
-                class="invoker"
+                class="summary"
                 role="button"
-                aria-controls="disclosure-content"
+                part="summary"
+                aria-controls="content"
                 aria-expanded="${x => x.expanded}"
             >
-                <slot name="start"></slot>
+                ${startSlotTemplate(options)}
                 <slot name="summary">${x => x.summary}</slot>
```

Comment:
Just responding to the note, I still think this attribute is a good affordance and ergonomic. If I just have a simple text string, I can pass it as an attribute without having to slot in an item.

Reply:
I was trying to figure out if there was a way to slot in some custom content that also binds to `x.summary`. I don't think there's a way to do that outside of the template itself, similar to the question I had on Discord about nesting templates. In that case, if you slotted something in the summary, it might be confusing that the attribute of the same name is ignored. Not sure the best way to guide around that, at least the documentation for this component I suppose.

Reply:
Separately, this is my current proposal for handling `label`. I'd like to see components that support a label have an attribute for providing that, seeing that labels are often simple small strings. Those would then bind into the `<label>` element, but could also be slotted in If someone needed an icon or custom content.

My same concern above applies here, which is why I was still thinking on this.

Reply:
> Separately, this is my current proposal for handling `label`. I'd like to see components that support a label have an attribute for providing that, seeing that labels are often simple small strings. Those would then bind into the `<label>` element, but could also be slotted in If someone needed an icon or custom content.
> 
> My same concern above applies here, which is why I was still thinking on this.

I think this would need a separate issue. In terms of label, *most* form elements can take a simple string because the default slot is for the label. This isn't the case for complex widgets like combobox and select because the default slot is for the slotted options or items. The consistent approach here, which was brought up by @eisenbergeffect in a prior PR of mine, would actually be to align to *that* approach, where everything has a "label" slot, but I wonder about consistency and convention over developer experience and intuitiveness. I'd like to not tightly couple the label work here because we should discuss that separately and get feedback from a broader audience about proposals to that.

Reply:
> I was trying to figure out if there was a way to slot in some custom content that also binds to `x.summary`. I don't think there's a way to do that outside of the template itself, similar to the question I had on Discord about nesting templates. In that case, if you slotted something in the summary, it might be confusing that the attribute of the same name is ignored. Not sure the best way to guide around that, at least the documentation for this component I suppose.

You can still bind to the attribute when slotting in if you use this in a template or if we were to add the slot as an "option" (then it would be treated as "default" slotted content which binds to the attribute. I'd prefer to not lose the ergonomic, so perhaps adding an affordance for this to be configured via "options" is the ideal there. Then when setting the options you're just passing the attribute as part of the view template being set.

```ts
html<Disclosure>`<custom-summary slot="summary">${x => x.summary}</custom-summary>`
```

Reply:
Agreed on handling labels as a separate issue.

Also agreed on Chris's last comment, that the summary wrapper should probably be an option. I'm still not finding a way to pass a template to a slot if the component just exists like
```
<fast-disclosure>
    <span slot="summary">Something the binds to x.summary here</span>
</fast-disclosure>